src/sql/src/normalize.rs

```diff
@@ -411,6 +412,22 @@ macro_rules! with_option_type {
     };
 }
 
+/// Ensures that the given set of options are empty, useful for validating that
+/// `WITH` options are all real, used options
+pub(crate) fn ensure_empty_options<V>(
+    with_options: &BTreeMap<String, V>,
+    r#for: &str,
```

Comment:
Using a raw ident is def nonstandard practice. Elsewhere in `sql` we call this sort of parameter `name` or `ctx`.